### PR TITLE
fix(etl): stock/traspasos COUNT crashes on None-valued driver row

### PR DESCRIPTION
**Pull request by alvarolobato**
## Summary

- Both `_count_expo` and `_count_traspasos` in `etl/sync/stock.py` called `int(next(iter(row.values())))` against an unaliased `SELECT COUNT(*)`. When the p4d driver returned a row whose first value was `None`, `int(None)` raised `int() argument must be a string, a bytes-like object or a real number, not 'NoneType'` and aborted the whole sync for those tables — visible as `Error` rows on the run-detail page.
- Fixed by adding an explicit `AS cnt` alias (consistent with every other test/module that reads a COUNT) and falling back to `0` on empty result or `None` value. COUNT is used for progress logging only; a missing count should not abort a sync that would otherwise succeed.
- Added 7 unit tests in `TestCountHelpers` covering the normal path, `None` value, empty result set, and the missing-alias fallback for both helpers.

## Why this happened

The original code had a comment acknowledging that "COUNT(*) key depends on the 4D driver version" but only handled the key-name variability, not the `None` value case. The delta queries (`WHERE FechaModifica > {d 'YYYY-MM-DD'}`) reached a state where the driver returned `{'some_key': None}` instead of an integer, and the whole pipeline step tipped over.

## Test plan

- [x] `ruff format` and `ruff check` clean on both files
- [x] `pytest etl/tests/test_sync_stock.py` — all 32 unit tests pass (3 integration tests skipped: no p4d locally)
- [ ] Next scheduled run (or manual `ps etl run`) should show `stock` and `traspasos` as `ok` rather than `Error` on the run-detail page

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: etl/tests/test_sync_stock.py [hunk 1]
```diff
@@ -246,6 +248,58 @@ def test_name_param_accepted(self):
...
 
 
+class TestCountHelpers:
+    """Unit tests for _count_expo / _count_traspasos — no real connection needed.
+
+    Regression coverage for the ``int() argument ... not 'NoneType'`` crash that
+    surfaced on stock/traspasos delta runs when the p4d driver returned a row
+    whose first value was ``None``.  The helpers are used for progress logging
+    only, so they must degrade to 0 instead of aborting the sync.
+    """
```
[the remark this answers is not in the document]
Comment (alvarolobato) [reply etl/tests/test_sync_stock.py:258]: Added test_count_traspasos_falls_back_to_first_value_when_alias_missing in 4467603 — symmetric with the _count_expo fallback test. PR description no longer drifts from the tests.

File: etl/tests/test_sync_stock.py [hunk 2]
```diff
@@ -246,6 +248,58 @@ def test_name_param_accepted(self):
...
+        with patch("etl.sync.stock.safe_fetch", return_value=[{"cnt": None}]):
+            assert _count_expo(conn, "") == 0
+
+    def test_count_expo_empty_result_returns_zero(self):
+        conn = MagicMock()
+        with patch("etl.sync.stock.safe_fetch", return_value=[]):
+            assert _count_expo(conn, "") == 0
+
+    def test_count_expo_falls_back_to_first_value_when_alias_missing(self):
+        """Driver quirk: if ``cnt`` key is absent, use the first populated value."""
```
[the remark this answers is not in the document]
Comment (alvarolobato) [reply etl/tests/test_sync_stock.py]: Reworded to 'first value in the row' in 4467603. The helper intentionally takes the first value verbatim and lets the trailing int(val) if val is not None else 0 handle the None case — so 'first populated value' was misleading.
